### PR TITLE
[JENKINS-35291] New version for BC 1.54

### PR DESCRIPTION
**Pull request by alvarolobato**
[JENKINS-35291](https://issues.jenkins-ci.org/browse/JENKINS-35291)
- New version for BC 1.54
- Added new tests for certificates

@reviewbybees esp @stephenconnolly @amuniz

### CODE REVIEW

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 1]
```diff
@@ -128,23 +143,51 @@ public static PEMEncodable decode(@Nonnull String pem) throws IOException, Unrec
...
+                }
+            } else if (object instanceof PKCS8EncryptedPrivateKeyInfo) {
+                if (passphrase != null) {
+                    InputDecryptorProvider dp = new JceOpenSSLPKCS8DecryptorProviderBuilder().build(passphrase);
+                    PKCS8EncryptedPrivateKeyInfo epk = (PKCS8EncryptedPrivateKeyInfo) object;
+                    return new PEMEncodable(kConv.getPrivateKey(epk.decryptPrivateKeyInfo(dp)));
+                } else {
+                    throw new UnrecoverableKeyException();
+                }
+            } else if (object instanceof PEMKeyPair) {
```
Comment (alvarolobato) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: Had to do all the nasty conversions to JCA objects conversions here 😞 . My planning was doing it in the toKeyPair, toPublicKey, etc, bu that would mean checked exceptions also on those methods.

**Comment by amuniz** [comment]
:bee: (after tests pass)

**Comment by andresrc** [comment]
:bee:

**Comment by stephenc** [comment]
The PR title is incorrect. It should reference a JENKINS issue not the internal CloudBees tracking JIRA issue.

Slightly alarmed to see the openssl package name creep in, but after inspecting the current code it seems that this should not introduce a code path that requires the openssl libs on the binary load path.

:bee:

**Comment by amuniz** [comment]
Test with a plugin that uses `bouncycastle-api` (1.0) by:
 * Upgrade `bouncycastle-api` to this snapshot
 * Run `mvn hpi:run -Djenkins.version=1.648`
 * It works fine

**Pull request merged**